src/integrate/integrate.ts

```diff
@@ -44,15 +44,29 @@ const syncToAdmina = async (source: Source, inputs: Record<string, string>) => {
       console.log('Getting Azure AD data...')
       const azureAdData = await AzureAdSource.fetchApps(inputs)
       console.log('Registering custom service...')
-      await PromisePool
-        .for(azureAdData)
-        .withConcurrency(2) // 並列数を5に制限
-        .process(async (app: AzureAdSource.AppInfo) => {
-          await AdminaDist.registerCustomService(
-            await AzureAdTransform.transformDataToAdmina(app),
-            inputs
-          )
+      try {
+        const { results, errors } = await PromisePool
+          .for(azureAdData)
+          .withConcurrency(5) // 並列数を5に制限
+          .process(async (app: AzureAdSource.AppInfo) => {
+            await AdminaDist.registerCustomService(
+              await AzureAdTransform.transformDataToAdmina(app),
+              inputs
+            )
+          })
+        errors.forEach(error => {
+          throw
+          if (error instanceof Error) {
+            throw new Error(error.message);
+          }
         })
+      } catch (error) {
+        if (error instanceof Error) {
+          throw new Error(error.message);
+        } else {
+          throw new Error(`Failed to register application data. :${error}`);
+        }
+      }
```

Comment:
I think this part is unnecessary.
You have a top lvl catch here already

https://github.com/moneyforward-i/admina-custom-service-action/blob/eecf724abbbd96b84e3bbe633bc1106a6a020f6a/src/index.ts#L44-L49

Also, Error contains message and stack, we only report the message, not the stack, which could be helpful with debugging, so in the top lvl try-catch I'd add an extra line before `if (error instanceof Error) {`

```
console.error('Failed due to error:', error);
```